zone.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 The Inet.Af AUTHORS. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !go1.19
+
+package netaddr
+
+import (
+	"runtime"
+	"sync"
+	"unsafe"
+)
+
+// zone is the IPv6 zone and its generation count to prevent the finalizer
+// from deleting weak references from the uniqZone map.
+type zone struct {
+	// name is the IPv6 zone.
+	// It is immutable.
+	name string
+
+	// gen is guarded by zmu and is incremented whenever this zone
+	// is returned.
+	gen int64
+}
+
+var (
+	z0    = (*zone)(nil)
+	z4    = new(zone)
+	z6noz = new(zone)
+)
+
+// zmu guards uniqZone, a weakref map of *zones by zoneName.
```

Comment:
Mention here that zmu also guards fields within zone. It's mentioned above, but is unusual enough that I think it deserves an xref here.

Reply:
Done.